lib/include/ert/ecl/smspec_node.hpp

```diff
@@ -56,13 +45,176 @@ typedef enum {ECL_SMSPEC_INVALID_VAR            =  0 ,
+                   int   lgr_i, int lgr_j , int lgr_k );
+    void common_init( ecl_smspec_var_type var_type_ , const char * keyword , const std::string& unit );
+    void set_num( const int grid_dims[3] , int num_);
+    void set_keyword( const std::string& keyword_ );
+    void set_flags();
+    void init_num( ecl_smspec_var_type var_type_);
+    void set_gen_keys( const char * key_join_string_);
+    void decode_R1R2( int * r1 , int * r2)  const;
+    void set_lgr_ijk( int lgr_i , int lgr_j , int lgr_k);
+
+
+    int cmp__(const smspec_node_type * node2) const;
+    int cmp_KEYWORD_WGNAME_NUM__(const smspec_node_type * node2) const;
+    static int cmp_KEYWORD_WGNAME_NUM( const smspec_node_type * node1, const smspec_node_type * node2) {
+      return node1->cmp_KEYWORD_WGNAME_NUM__(node2);
+    }
+    int cmp_KEYWORD_WGNAME_LGR__( const smspec_node_type * node2) const;
+    static int cmp_KEYWORD_WGNAME_LGR( const smspec_node_type * node1, const smspec_node_type * node2) {
+      return node1->cmp_KEYWORD_WGNAME_LGR__(node2);
+    }
+    int cmp_KEYWORD_WGNAME_LGR_LGRIJK__( const smspec_node_type * node2) const;
+    static int cmp_KEYWORD_WGNAME_LGR_LGRIJK( const smspec_node_type * node1, const smspec_node_type * node2) {
+      return node1->cmp_KEYWORD_WGNAME_LGR_LGRIJK__(node2);
+    }
+    int cmp_KEYWORD_WGNAME__( const smspec_node_type * node2) const;
+    static int cmp_KEYWORD_WGNAME( const smspec_node_type * node1, const smspec_node_type * node2) {
+      return node1->cmp_KEYWORD_WGNAME__(node2);
+    }
+    static bool equal_MISC( const smspec_node_type * node1, const smspec_node_type * node2) {
+      return node1->keyword == node2->keyword;
+    }
+    int cmp_MISC__( const smspec_node_type * node2) const;
+    static int cmp_MISC( const smspec_node_type * node1, const smspec_node_type * node2) {
+      return node1->cmp_MISC__(node2);
+    }
+    int cmp_LGRIJK__( const smspec_node_type * node2) const;
+    static int cmp_LGRIJK( const smspec_node_type * node1, const smspec_node_type * node2) {
+      return node1->cmp_LGRIJK__(node2);
+    }
+    int cmp_KEYWORD_LGR_LGRIJK__( const smspec_node_type * node2) const;
+    static int cmp_KEYWORD_LGR_LGRIJK( const smspec_node_type * node1, const smspec_node_type * node2) {
+      return node1->cmp_KEYWORD_LGR_LGRIJK__(node2);
+    }
+    int cmp_KEYWORD_NUM__( const smspec_node_type * node2) const;
+    static int cmp_KEYWORD_NUM( const smspec_node_type * node1, const smspec_node_type * node2) {
+      return node1->cmp_KEYWORD_NUM__(node2);
+    }
+    static int cmp_KEYWORD( const smspec_node_type * node1, const smspec_node_type * node2) {
+      return node1->keyword.compare(node2->keyword);
+    }
+    int cmp_key1__( const smspec_node_type * node2) const;
+    static int cmp_key1( const smspec_node_type * node1, const smspec_node_type * node2) {
+      return node1->cmp_key1__(node2);
+    }
+
+  public:
+
+    UTIL_TYPE_ID_DECLARATION;
```

Comment:
This should *really* be private!

Reply:
It can't, since it is used by UTIL_TYPE_ID_DECLARATION.